src/EGraphs/saturation.jl

```diff
@@ -144,14 +99,12 @@ end
 Instantiate argument for dynamic rule application in e-graph
 """
 function instantiate_actual_param!(bindings, g::EGraph, i)
+  const_hash = v_pair_last(bindings[i])
+  const_hash == 0 || return get_constant(g, const_hash)
+
   ecid = v_pair_first(bindings[i])
-  literal_position = reinterpret(Int, v_pair_last(bindings[i]))
   ecid <= 0 && error("unbound pattern variable")
   eclass = g[ecid]
-  if literal_position > 0
-    @assert !v_isexpr(eclass[literal_position])
-    return get_constant(g, v_head(eclass[literal_position]))
-  end
```

Comment:
Why is literals support removed here?

Reply:
The matchers do not return literal positions anymore (see line 330 in ematch_compiled.jl yield_expr()) . Instead the hash is returned even for literals and they are handled in lines 102 and 103 instead.

Reply:
Nice @gkronber thanks :) Can we handle this change and the schedulers in two different PRs?